src/app/CommandSender.h

```diff
@@ -229,6 +158,74 @@ class CommandSender final : public Messaging::ExchangeDelegate
         virtual void OnDone(CommandSender * apCommandSender) = 0;
     };
 
+    /**
+     * @brief Legacy callbacks for CommandSender
+     */
+    class Callback
```

Comment:
This should go after all public ReadClient APIs, not just after ExtendableCallback, no?

Maybe even in a separate public: section somewhere after all the private: stuff.

Basically, make it so it does not pop up and confuse people.

Reply:
@bzbarsky-apple  (Low priority PR)

I am not sure where we discussed that I should just move this into it's own header file, but I recall having that discussion somewhere. Anyways I have moved legacy callback into it's own header file now and this PR is ready for another review